.buildkite/scripts/publish.py

```diff
@@ -47,5 +43,7 @@ def publish(project_name):
         print(f"Nothing in this patch affects {projects}, so stopping.")
         sys.exit(0)
 
-    for p in projects:
-        publish(p)
+    for project_name in projects:
+        run_build_script(
```

Comment:
Does this propagate the environment variables to the task in docker? Causeto publish it needs the BUILDKITE_BUILD_NUMBER otherwise it will put a build number 0, which is only supposed to happen locally. The previous YAML way had `propagate-environment: true` for this reason

Reply:
Yes: https://github.com/wellcomecollection/catalogue-pipeline/blob/main/builds/run_sbt_task_in_docker.sh#L32